File: Java/Data Structures & Algorithms Module - Java/Week11/LinearSearch/LinearSearch2.java
/*
LinearSearch2.java
Billy Cussen
16/12/2020
*/

import java.util.List;
import java.util.ArrayList;

public class LinearSearch2<T>{

    public static <T extends Comparable<T>> boolean linearSearchGenerics(List<T> myList, T key){
        for(int i = 0; i < myList.size();i++){
            if(myList.get(i).compareTo(key)==0){
                return true;
            }
        }
        return false;
    }

    public static boolean linearSearchArray(int[] myArray, int key){
        for(int i = 0; i < myArray.length; i++){
            if(myArray[i]==key){
                return true;
            }
        }
        return false;
    }

    public static void main(String [] args){

        List<Integer> myList = new ArrayList<Integer>();
        myList.add(1);
        myList.add(2);
        myList.add(3);
        myList.add(4);
        myList.add(5);
        myList.add(6);
        int[] myArray = new int[]{1,2,3,4,5,6,7,8,9,10};

        System.out.println("Does List contain 2: "+linearSearchGenerics(myList, 2));
        System.out.println("Does List contain 200: "+linearSearchGenerics(myList, 200));
        System.out.println("Does List contain 4: "+linearSearchGenerics(myList, 4));
        System.out.println("Does List contain 10: "+linearSearchGenerics(myList, 10));

        System.out.println("Does Array contain 2: "+linearSearchArray(myArray, 2));
        System.out.println("Does Array contain 200: "+linearSearchArray(myArray, 200));
        System.out.println("Does Array contain 4: "+linearSearchArray(myArray, 4));
        System.out.println("Does Array contain 60: "+linearSearchArray(myArray, 60));
    }

}

File: Python/Data Structures & Algorithms Module - Python/Week11/LinearSearch/LinearSearch2.py
"""
LinearSearch2.py
Billy Cussen
16/12/2020
"""

def linearSearch2(myArray, key):
    for i in range(len(myArray)):
        if myArray[i] == key:
            return True
    return False


myArr = [1,2,3,4,5,6,7,8,9,10]

print("Does Array Contain 2: "+str(linearSearch2(myArr, 2)))
print("Does Array Contain 4: "+str(linearSearch2(myArr, 4)))
print("Does Array Contain 6: "+str(linearSearch2(myArr, 6)))
print("Does Array Contain 8: "+str(linearSearch2(myArr, 8)))
print("Does Array Contain 10: "+str(linearSearch2(myArr, 10)))
print("Does Array Contain 12: "+str(linearSearch2(myArr, 12)))
print("Does Array Contain 14: "+str(linearSearch2(myArr, 14)))
    